src/integrationTest/kotlin/com/pubnub/api/integration/SubscribeIntegrationTests.kt

```diff
@@ -115,6 +117,55 @@ class SubscribeIntegrationTests : BaseIntegrationTest() {
         success.listen()
     }
 
+    @Test
+    fun testTwoAndMoreConsecutiveSubscribeCallFirstWithTimeTokenShouldReceiveMessages() {
+        val countDownLatch = CountDownLatch(3)
+
+        // make two pubnub instances
+        val pubnub1 = PubNub(getBasicPnConfiguration())
+        val pubnub2 = PubNub(getBasicPnConfiguration())
+
+        // create a channel
+        val channel01 = randomChannel()
+
+        // create messages
+        val expectedMessage01 = randomValue()
+        val expectedMessage02 = randomValue()
+        val expectedMessage03 = randomValue()
+
+        // send a message from user 1
+        pubnub1.publish(channel01, expectedMessage01).sync()
+
+        // send a message from user 2
+        pubnub2.publish(channel01, expectedMessage02).sync()
+
+        // get timestamp for 10 minutes ago
+        val timeToken10minAgo = (System.currentTimeMillis() - 600_000L) * 10_000L
+
+        pubnub1.addListener(object : SubscribeCallback() {
+            override fun status(pubnub: PubNub, pnStatus: PNStatus) {
+                // nothing here
+            }
+
+            override fun message(pubnub: PubNub, pnMessageResult: PNMessageResult) {
+                println("-=message ${pnMessageResult.message}") // <-- here I am able to get msgFromUser1ToChannel01 and msgFromUser2ToChannel01
+                assertTrue(listOf(expectedMessage01, expectedMessage02, expectedMessage03).contains(pnMessageResult.message.asString))
+                countDownLatch.countDown()
+            }
+        })
+
+        pubnub1.subscribe(channels = listOf(channel01), withTimetoken = timeToken10minAgo)
+        pubnub1.publish(channel01, expectedMessage03).sync()
```

Comment:
is this publish necessary here to trigger the error condition (before the fix?) 
I'm worried that it might introduce an unnecessary delay between subscribes, if it's not needed consider removing

Reply:
I tested before and without the fixed code  a presence of this additional publish doesn't effect result.
Additionally I would like to test that subscribe call that follows this publish doesn't cause the message from this publish being not deliver to listener.